packages/core/utils/src/common/define-config.ts

```diff
@@ -0,0 +1,127 @@
+ */
+export function defineConfig(config: Partial<ConfigModule> = {}): ConfigModule {
+  const { http, ...restOrProjectConfig } = config.projectConfig || {}
+
+  /**
+   * The defaults to use for the project config. They are shallow merged
+   * with the user defined config. However,
+   */
+  const projectConfig: ConfigModule["projectConfig"] = {
+    databaseUrl: process.env.DATABASE_URL || DEFAULT_DATABASE_URL,
+    http: {
+      storeCors: process.env.STORE_CORS || DEFAULT_STORE_CORS,
+      adminCors: process.env.ADMIN_CORS || DEFAULT_ADMIN_CORS,
+      authCors: process.env.AUTH_CORS || DEFAULT_ADMIN_CORS,
+      jwtSecret: process.env.JWT_SECRET || DEFAULT_SECRET,
+      cookieSecret: process.env.COOKIE_SECRET || DEFAULT_SECRET,
+      ...http,
+    },
+    ...restOrProjectConfig,
+  }
+
+  /**
+   * The defaults to use for the admin config.  They are shallow merged
+   * with the user defined config
+   */
+  const admin: ConfigModule["admin"] = {
+    backendUrl: DEFAULT_ADMIN_URL,
+    ...config.admin,
+  }
+
+  /**
+   * The defaults to use for the feature flags config. They are shallow merged
+   * with the user defined config
+   */
+  const featureFlags: ConfigModule["featureFlags"] = {
+    medusa_v2: true,
+    ...config.featureFlags,
+  }
+
+  /**
+   * The default set of modules to always use. The end user can swap
+   * the modules by providing an alternate implementation via their
+   * config. But they can never remove a module from this list.
+   */
+  const modules: ConfigModule["modules"] = {
+    [Modules.CACHE]: true,
+    [Modules.EVENT_BUS]: true,
+    [Modules.AUTH]: true,
+    [Modules.WORKFLOW_ENGINE]: true,
+    [Modules.STOCK_LOCATION]: true,
+    [Modules.INVENTORY]: true,
+    [Modules.PRODUCT]: true,
+    [Modules.PRICING]: true,
+    [Modules.PROMOTION]: true,
+    [Modules.CUSTOMER]: true,
+    [Modules.SALES_CHANNEL]: true,
+    [Modules.CART]: true,
+    [Modules.REGION]: true,
+    [Modules.API_KEY]: true,
+    [Modules.STORE]: true,
+    [Modules.TAX]: true,
+    [Modules.CURRENCY]: true,
+    [Modules.PAYMENT]: true,
+    [Modules.ORDER]: true,
+    [Modules.USER]: {
+      resolve: "@medusajs/user",
+      options: {
+        jwt_secret: process.env.JWT_SECRET ?? DEFAULT_SECRET,
+      },
+    },
+    [Modules.FILE]: {
+      resolve: "@medusajs/file",
+      options: {
+        providers: [
+          {
+            resolve: "@medusajs/file-local-next",
+            options: {
+              config: {
+                local: {},
+              },
+            },
+          },
+        ],
+      },
+    },
+    [Modules.FULFILLMENT]: {
+      resolve: "@medusajs/fulfillment",
+      options: {
+        providers: [
+          {
+            resolve: "@medusajs/fulfillment-manual",
+            options: {
+              config: {
+                manual: {},
+              },
+            },
+          },
+        ],
+      },
+    },
+    ...config.modules,
+  }
```

Comment:
```suggestion
   const modules: ConfigModule["modules"] = {
    ...Object.keys(MODULE_PACKAGE_NAMES).reduce((acc, key) => {
      acc[key] = true;
      return acc;
    }, {}),
    [Modules.USER]: {
      resolve: "@medusajs/user",
      options: {
        jwt_secret: process.env.JWT_SECRET ?? DEFAULT_SECRET,
      },
    },
    [Modules.FILE]: {
      resolve: "@medusajs/file",
      options: {
        providers: [
          {
            resolve: "@medusajs/file-local-next",
            options: {
              config: {
                local: {},
              },
            },
          },
        ],
      },
    },
    [Modules.FULFILLMENT]: {
      resolve: "@medusajs/fulfillment",
      options: {
        providers: [
          {
            resolve: "@medusajs/fulfillment-manual",
            options: {
              config: {
                manual: {},
              },
            },
          },
        ],
      },
    },
    ...config.modules,
  }
```

Wdyt? Take it or leave it :D, I am just suggesting that such that if we add a new module then it will be present in most cases. But otherwise I am fine with the current one

Reply:
Yeah, in one instance I like it. But in another one, I think let it be a explicit of explicit module names. Not sure, its hard to decide 😬